hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -750,24 +767,49 @@ private HoodieTimeline getInflightTimelineExcludeCompactionAndClustering(HoodieT
   }
 
   /**
-   * Cleanup all pending commits.
+   * Clean up all failed commits.
+=======
+   * Rollback all failed commits.
+>>>>>>> Adding configs to make failed writes eager/lazy
+=======
+   * Rollback all failed writes.
+>>>>>>> Refactored code, removed parallel writing capability
    */
-  private void rollbackPendingCommits() {
+  public void rollbackFailedWrites() {
     HoodieTable<T, I, K, O> table = createTable(config, hadoopConf);
-    HoodieTimeline inflightTimeline = getInflightTimelineExcludeCompactionAndClustering(table);
-    List<String> commits = inflightTimeline.getReverseOrderedInstants().map(HoodieInstant::getTimestamp)
-        .collect(Collectors.toList());
-    for (String commit : commits) {
-      if (HoodieTimeline.compareTimestamps(commit, HoodieTimeline.LESSER_THAN_OR_EQUALS,
+    List<String> instantsToRollback = getInstantsToRollback(table);
+    for (String instant : instantsToRollback) {
+      if (HoodieTimeline.compareTimestamps(instant, HoodieTimeline.LESSER_THAN_OR_EQUALS,
           HoodieTimeline.FULL_BOOTSTRAP_INSTANT_TS)) {
-        rollBackInflightBootstrap();
+        rollbackFailedBootstrap();
         break;
       } else {
-        rollback(commit);
+        rollback(instant);
       }
     }
   }
 
+  private List<String> getInstantsToRollback(HoodieTable<T, I, K, O> table) {
+    if (config.getFailedWritesCleanPolicy().isEager()) {
+      HoodieTimeline inflightTimeline = table.getMetaClient().getCommitsTimeline().filterPendingExcludingCompaction();
+      return inflightTimeline.getReverseOrderedInstants().map(HoodieInstant::getTimestamp)
+          .collect(Collectors.toList());
+    } else if (!config.getFailedWritesCleanPolicy().isEager()) {
```

Comment:
would nt `HoodieFailedWritesCleaningPolicy.NEVER` also pass the check for `!eager` ?

Reply:
moved the else if conditions